geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientRegistrationMetadata.java

```diff
@@ -0,0 +1,193 @@
+        dataOutputStream = unversionedDataOutputStream;
+      }
+
+      // Read and ignore the reply code. This is used on the client to server
+      // handshake.
+      dataInputStream.readByte(); // replyCode
+
+      // Read the ports and throw them away. We no longer need them
+      int numberOfPorts = dataInputStream.readInt();
+      for (int i = 0; i < numberOfPorts; i++) {
+        dataInputStream.readInt();
+      }
+
+      getAndValidateClientProxyMembershipID();
+
+      if (getAndValidateClientConflation()) {
+        clientCredentials =
+            Handshake.readCredentials(dataInputStream, dataOutputStream,
+                cache.getDistributedSystem(), cache.getSecurityService());
+
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  ClientProxyMembershipID getClientProxyMembershipID() {
+    return clientProxyMembershipID;
+  }
+
+  byte getClientConflation() {
+    return clientConflation;
+  }
+
+  Properties getClientCredentials() {
+    return clientCredentials;
+  }
+
+  Version getClientVersion() {
+    return clientVersion;
+  }
+
+  DataOutputStream getDataOutputStream() {
+    return dataOutputStream;
+  }
+
+  private boolean getAndValidateClientVersion(final Socket socket,
+      final DataInputStream dataInputStream, final DataOutputStream dataOutputStream)
+      throws IOException {
+    short clientVersionOrdinal = Version.readOrdinal(dataInputStream);
+
+    try {
+      clientVersion = Version.fromOrdinal(clientVersionOrdinal, true);
+      if (isVersion57orOlder(clientVersion)) {
+        throw new IOException(new UnsupportedVersionException(clientVersionOrdinal));
+      }
+      if (logger.isDebugEnabled()) {
+        logger.debug("{}: Registering client with version: {}", this, clientVersion);
+      }
+    } catch (UnsupportedVersionException e) {
+      UnsupportedVersionException unsupportedVersionException = e;
+      SocketAddress socketAddress = socket.getRemoteSocketAddress();
+
+      if (socketAddress != null) {
+        String sInfo = " Client: " + socketAddress.toString() + ".";
+        unsupportedVersionException = new UnsupportedVersionException(e.getMessage() + sInfo);
+      }
+
+      logger.warn(
+          "CacheClientNotifier: Registering client version is unsupported.  Error details: ",
+          unsupportedVersionException);
+
+      socketMessageWriter.writeException(dataOutputStream,
+          CommunicationMode.UnsuccessfulServerToClient.getModeNumber(),
+          unsupportedVersionException, null);
+
+      return false;
+    }
+
+    return true;
+  }
+
+  private boolean doesClientSupportExtractOverrides() {
+    return clientVersion.compareTo(Version.GFE_603) >= 0;
+  }
+
+  private boolean oldClientRequiresVersionedStreams(final Version clientVersion) {
+    return Version.CURRENT.compareTo(clientVersion) > 0;
+  }
+
+  private boolean isVersion57orOlder(final Version clientVersion) {
+    return Version.GFE_57.compareTo(clientVersion) > 0;
```

Comment:
I think this method should be named isOlderThanVersion57

Reply:
You are correct - fixed that, thanks!